### DISCUSSION TITLE
Improve privacy options to allow a setting for visibility only to descendants

### DISCUSSION THREAD

**Discussion by vteague** [fisharebest/webtrees#5018] [Ideas]
Hi, I'm working on a project with a very large family tree (>1400 people, including living and dead), and we'd like to introduce a setting in which some individuals (specifically, deceased individuals) are visible only to their direct descendants.

We saw that there was a previous PR with this feature (#905), which wasn't merged, but your proposed solution - splitting the tree into two pieces - doesn't work for our use case. We have a very large tree, with multiple different people who need different views depending on their ancestry - splitting it into two doesn't help. 

I'd be happy to do the implementation and write some test cases, possibly adopting the previous code form the PR mentioned above, and of course we'd all be happy to have a discussion about how best to go about it. My preliminary idea was to extend deceased individuals' privacy settings by some length of time, and then edit the code for allowing visibility to near relatives to add an alternative that just calculated visibility for descendants.

**Comment by Manja-S** [comment]
I'd like an option to set this for living people as well.

**Comment by hartenthaler** [comment]
Is there any progress in that topic? I would like to integrate such a function into my module hh_privacy_assistant.
